lib/std/Thread.zig

```diff
@@ -572,6 +754,82 @@ pub fn getCurrentThreadId() u64 {
     }
 }
 
+fn testThreadName(thread: *Thread) !void {
+    // Generate a name of the maximum allowed length.
+    comptime const max_length_name = blk: {
+        var tmp: [maxNameLen + 1]u8 = undefined;
+        std.mem.set(u8, &tmp, 'b');
+        tmp[maxNameLen] = 0;
+        break :blk tmp[0..maxNameLen :0];
+    };
+
+    comptime const testCases = &[_][]const u8{
+        "mythread",
+        max_length_name,
+    };
+
+    inline for (testCases) |tc| {
+        try thread.setName(tc);
+
+        const name = try thread.getName(std.testing.allocator);
+        if (name) |value| {
+            defer std.testing.allocator.free(value);
+            std.testing.expectEqualStrings(tc, value);
+        }
+    }
+}
+
+fn testThreadNameUnsupported(thread: *Thread) !void {
+    const res = thread.setName("foobar");
+    std.testing.expectError(error.Unsupported, res);
+
+    const res2 = thread.getName(std.testing.allocator);
+    std.testing.expectError(error.Unsupported, res2);
+}
+
+test "Thread: setName, getName" {
+    if (builtin.single_threaded) return error.SkipZigTest;
+
+    const Context = struct {
+        done: std.atomic.Bool = std.atomic.Bool.init(false),
+
+        pub fn run(ctx: *@This()) !void {
+            while (!ctx.done.load(.SeqCst)) {
+                std.time.sleep(5 * std.time.ns_per_ms);
+            }
+        }
+    };
+
+    var context = Context{};
+    var thread = try spawn(Context.run, &context);
+
+    switch (std.Target.current.os.tag) {
+        .macos, .ios, .watchos, .tvos => {
+            const res = thread.setName("foobar");
+            std.testing.expectError(error.Unsupported, res);
+        },
+        .windows => if (std.Target.current.os.isAtLeast(.windows, .win10_rs1)) |res| {
```

Comment:
`isAtLeast` returns `null` if a runtime-check is required, I'd simply run `testThreadName` and catch `unsupported`, in that case you can return `SkipZigTest` as that's not an error condition.